src/Confluent.Kafka/Consumer.cs

```diff
@@ -87,7 +85,15 @@ public class Consumer<TKey, TValue> : IDisposable
                 ValueDeserializer = (IDeserializer<TValue>)new NullDeserializer();
             }
 
-            consumer = new Consumer(config);
+            var adjustedConfig1 = KeyDeserializer.Configure(config, true);
+            var adjustedConfig2 = ValueDeserializer.Configure(config, false);
+
+            consumer = new Consumer(
+                config.Where(item => 
+                    adjustedConfig1.Any(ci => ci.Key == item.Key) &&
```

Comment:
The (de)serializers could misbehave, but I don't see it as a real problem or any way of avoiding it except by maintaining a list of librdkafka properties, which I don't think is worth the overhead. The fundamental problem is we don't know which properties are librdkafka properties - I don't think thinking about this backwards helps.

will add a comment.